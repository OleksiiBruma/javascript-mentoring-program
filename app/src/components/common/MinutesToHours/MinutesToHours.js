import {Component} from "react";

class MinutesToHours extends Component {
  convert = seconds =>{
      return seconds ? new Date(parseInt(seconds) * 1000).toISOString().substr(11, 8) : null;
  } 
  render() {
    return this.props.render(this.convert);
  }
}

export default MinutesToHours;
