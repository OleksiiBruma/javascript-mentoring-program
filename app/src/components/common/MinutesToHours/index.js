import MinutesToHours from "./MinutesToHours";

export default MinutesToHours