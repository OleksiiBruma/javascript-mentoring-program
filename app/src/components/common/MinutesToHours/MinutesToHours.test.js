import renderer from "react-test-renderer";
import React from "react";
import MinutesToHours from "../MinutesToHours/MinutesToHours";

it("return seconds converted to hours", () => {
  const component = renderer.create(<MinutesToHours render={(convert)=>convert(3333)}/>);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

