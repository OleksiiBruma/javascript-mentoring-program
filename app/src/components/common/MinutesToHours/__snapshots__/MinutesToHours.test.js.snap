// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`return seconds converted to hours 1`] = `"00:55:33"`;
